crates/cache/src/utils.rs

```diff
@@ -88,3 +88,191 @@ fn collect_table_names(plan: &LogicalPlan, table_names: &mut HashSet<String>) {
         collect_table_names(input, table_names);
     });
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use arrow::datatypes::{DataType, Field, Schema};
+    use datafusion::datasource::MemTable;
+    use datafusion::execution::config::SessionConfig;
+    use datafusion::execution::context::SessionContext;
+    use std::collections::HashSet;
+
+    async fn parse_sql_to_logical_plan(sql: &str) -> LogicalPlan {
+        let ctx = create_session_context();
+
+        let plan = &ctx
+            .state()
+            .create_logical_plan(sql)
+            .await
+            .expect("Should parse SQL to logical plan");
+
+        plan.clone()
+    }
+
+    #[tokio::test]
+    async fn test_collect_table_names_system_query_describe() {
+        let sql = "describe customer";
+        let logical_plan = parse_sql_to_logical_plan(sql).await;
+
+        let table_names = get_logical_plan_input_tables(&logical_plan);
+
+        let expected: HashSet<String> = vec![].into_iter().collect();
+        assert_eq!(table_names, expected);
+    }
```

Comment:
nit. Most of the tests are repetitive pattern, you can use a for-loop to put all tests into one to have a table-based test style.

Reply:
Thank you Yong!